db/seeds.rb

```diff
@@ -1,436 +1,175 @@
-      user.confirmed_at = Time.zone.now
-      create_data_protection_confirmation(user)
-    end
+    stock_owner1_user = find_or_create_user(stock_owner1, "stock_owner1_dpo@example.com", "Stock owner 1", :data_coordinator)
+    stock_owner2_user = find_or_create_user(stock_owner2, "stock_owner2_dpo@example.com", "Stock owner 2", :data_coordinator)
 
-    pp "Seeded dummy users"
-  end
+    managing_agent1_user = find_or_create_user(managing_agent1, "managing_agent1_dpo@example.com", "Managing agent 1", :data_coordinator)
+    managing_agent2_user = find_or_create_user(managing_agent2, "managing_agent2_dpo@example.com", "Managing agent 2", :data_coordinator)
 
-  if (Rails.env.development? || Rails.env.review?) && SalesLog.count.zero?
-    SalesLog.find_or_create_by!(
-      assigned_to: support_user,
-      owning_organisation: org,
-      managing_organisation: org,
-      saledate: Time.zone.today,
-      purchid: "1",
-      ownershipsch: 1,
-      type: 2,
-      jointpur: 1,
-      jointmore: 1,
-    )
+    provider_owner1 = find_or_create_user(standalone_owns_stock, "provider.owner1@example.com", "Provider Owns Stock", :data_provider)
+    coordinator_owner1 = find_or_create_user(standalone_owns_stock, "coordinator.owner1@example.com", "Coordinator Owns Stock", :data_coordinator)
 
-    SalesLog.find_or_create_by!(
-      assigned_to: support_user,
-      owning_organisation: org,
-      managing_organisation: org,
-      saledate: Time.zone.today,
-      purchid: "1",
-      ownershipsch: 2,
-      type: 9,
-      jointpur: 1,
-      jointmore: 1,
-    )
+    find_or_create_user(standalone_no_stock, "provider.nostock@example.com", "Provider No Stock", :data_provider)
+    find_or_create_user(standalone_no_stock, "coordinator.nostock@example.com", "Coordinator No Stock", :data_coordinator)
 
-    SalesLog.find_or_create_by!(
-      assigned_to: support_user,
-      owning_organisation: org,
-      managing_organisation: org,
-      saledate: Time.zone.today,
-      purchid: "1",
-      ownershipsch: 3,
-      type: 10,
-      companybuy: 1,
-    )
+    if Scheme.count.zero?
+      beulahside = FactoryBot.create(:scheme, service_name: "Beulahside Care", owning_organisation: mhclg)
+      abdullah = FactoryBot.create(:scheme, service_name: "Abdullahview Point", owning_organisation: mhclg)
 
-    pp "Seeded a sales log of each type"
-  end
+      FactoryBot.create(:location, scheme: beulahside, name: "Rectory Road", postcode: "CU193AA", location_code: "E09000033", location_admin_district: "Westminster", type_of_unit: 4, mobility_type: "N")
+      FactoryBot.create(:location, scheme: beulahside, name: "Smithy Lane", postcode: "DM25ODC", location_code: "E09000033", location_admin_district: "Westminster", type_of_unit: 1, mobility_type: "W")
+      FactoryBot.create(:location, scheme: abdullah, name: "Smithy Lane", postcode: "YX130WP", location_code: "E09000033", location_admin_district: "Westminster", type_of_unit: 2, mobility_type: "W")
```

Comment:
These match the existing locations / schemes - I add additional with some randomness below.